container.go

```diff
@@ -750,3 +774,45 @@ func (c *Container) detachDevices() error {
 
 	return nil
 }
+
+func (c *Container) addResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config != nil {
+		cType := ContainerType(c.config.Annotations[vcAnnotations.ContainerTypeKey])
+		if cType == PodSandbox {
+			virtLog.Debugf("Cannot hot add vCPUs to sandbox containers")
+			return nil
+		}
+
+		vCPUs := c.config.Resources.VCPUs
```

Comment:
Ok now I understand why you have embedded `Resources` into `ContainerResources`.
But I still think you should not need this since you have cpu quota and period, which should be enough to determine the numbers of vCPUs needed for this container, right ??